File: smp2017-Python-maulik/cs50/pset1/credit.c
#include <stdio.h>

int main(void)
{
	unsigned long long int ccn,smpl;
	int cnt=1,sum=0,i,n;

	printf("Enter your credit card number :");
	scanf("%lld",&ccn);

	smpl=ccn;

	while(smpl>0)
	{
		i=smpl%10;
		n=i;
		if(cnt%2==0)
		{
			n=2*i;
			while(n>0)
			{
				sum+=n%10;
				n=n/10;
			}
		}
		else
		{
			sum+=i;
		}
		cnt++;
		smpl=smpl/10;
	}

	if(sum%10==0)
		printf("\nYour Credit card is Legit.\n");
	else
		printf("\nYou cheated.Invalid credit card!\n");
}

File: smp2017-Python-maulik/cs50/pset6/credit.py
ccn=int(input('Enter your credit card number : '))

smpl=ccn
cnt=1
s=0

while smpl > 0 :
    i=smpl%10

    if(cnt%2==0):
        n=2*i
        while n > 0:
            s+=n%10
            n=n//10
    else:
        s+=i

    cnt+=1;
    smpl=smpl//10


if(s%10==0) :
    print('Your Credit card is legit.')
else :
    print('You cheated.Invlaid credit card!')
